toolkit/tools/pkg/isomakerlib/isomaker.go

```diff
@@ -47,20 +47,25 @@ type IsoMaker struct {
 	imageNameBase      string               // Base name of the ISO to generate (no path, and no file extension).
 	imageNameTag       string               // Optional user-supplied tag appended to the generated ISO's name.
 
-	isoMakerCleanUpTasks []func() // List of clean-up tasks to perform at the end of the ISO generation process.
+	isoMakerCleanUpTasks []func() error // List of clean-up tasks to perform at the end of the ISO generation process.
 }
 
 // NewIsoMaker returns a new ISO maker.
-func NewIsoMaker(unattendedInstall bool, baseDirPath, buildDirPath, releaseVersion, resourcesDirPath, configFilePath, initrdPath, isoRepoDirPath, outputDir, imageNameTag string) *IsoMaker {
+func NewIsoMaker(unattendedInstall bool, baseDirPath, buildDirPath, releaseVersion, resourcesDirPath, configFilePath, initrdPath, isoRepoDirPath, outputDir, imageNameTag string) (*IsoMaker, error) {
```

Comment:
In cases of more errors (or even with one, recently) we're tring to use named output arguments:
```suggestion
func NewIsoMaker(unattendedInstall bool, baseDirPath, buildDirPath, releaseVersion, resourcesDirPath, configFilePath, initrdPath, isoRepoDirPath, outputDir, imageNameTag string) (isoMake *IsoMaker, err error) {
```

This may require a few changes below.

Reply:
@PawelWMS - used named output arguments where it makes sense.